tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/team/mailboxes/TeamMailboxAutocompleteCallback.scala

```diff
@@ -0,0 +1,26 @@
+package com.linagora.tmail.james.jmap.team.mailboxes
+
+import com.linagora.tmail.james.jmap.contact.{ContactFields, EmailAddressContactSearchEngine}
+import com.linagora.tmail.team.{TeamMailbox, TeamMailboxCallback}
+import javax.inject.Inject
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.SMono
+
+class TeamMailboxAutocompleteCallback @Inject()(contactSearchEngine: EmailAddressContactSearchEngine) extends TeamMailboxCallback {
+
+  def teamMailboxAdded(teamMailbox: TeamMailbox): Publisher[Void] = {
+    def indexDomainContact(): SMono[Unit] =
```

Comment:
Q: Why do we need an extracted variable for it?
More readable?

Reply:
I should shorten this actually.